pkg/deployments/tasks/20230403-timelock-authorizer/input.ts

```diff
@@ -0,0 +1,79 @@
+import hre from 'hardhat';
+import { Contract } from 'ethers';
+import { getForkedNetwork } from '../../src/test';
+import Task, { TaskMode } from '../../src/task';
+import { DelayData, RoleData } from './input/types';
+import { ANY_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+
+const Authorizer = new Task('20210418-authorizer', TaskMode.READ_ONLY);
+const AuthorizerAdaptorEntrypoint = new Task('20221124-authorizer-adaptor-entrypoint', TaskMode.READ_ONLY);
+
+export type TimelockAuthorizerDeployment = {
+  Authorizer: string;
+  AuthorizerAdaptorEntrypoint: string;
+  Root: string;
+  Roles: RoleData[];
+  Granters: RoleData[];
+  Revokers: RoleData[];
+  ExecuteDelays: DelayData[];
+  GrantDelays: DelayData[];
+  TRANSITION_START_BLOCK: number;
+  TRANSITION_END_BLOCK: number;
+};
+
+export async function getOnChainRoles(roles: RoleData[], start: number, end: number): Promise<RoleData[]> {
+  const OldAuthorizerTask = new Task('20210418-authorizer', TaskMode.READ_ONLY);
+  const oldAuthorizerAddress = OldAuthorizerTask.output({ network: getForkedNetwork(hre) }).Authorizer;
```

Comment:
While this is correct for fork tests (because we want the forked network, e.g. mainnet, and not the real network, e.g. hardhat), it will fail in a real deployment (because we're not forking).